### PR TITLE
Casting spells; castingmanager

### PR DESCRIPTION
**Pull request by Thorvers**

### CODE REVIEW

**Review by Sven-dev — APPROVED** [review]

File: RPG Wizardry/Assets/Scripts/Avatar/Combat/CastingManager.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+{
+    [RequireComponent(typeof(InputState))]
+    public class CastingManager : MonoBehaviour
+    {
+        #region Variables
+        #region Public
+        /// <summary>
+        /// Prototype projectile; fill this with selected spell later
+        /// </summary>
+        public Projectile projectilePrefab;
```
Comment (frankvHoof93) [inline RPG Wizardry/Assets/Scripts/Avatar/Combat/CastingManager.cs:16]: is public set required?

File: RPG Wizardry/Assets/Scripts/Avatar/Combat/CastingManager.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        }
+
+        /// <summary>
+        /// Cooldown state during which no spells may be cast
+        /// </summary>
+        /// <param name="coolSeconds">Seconds the cooldown should remain active</param>
+        /// <returns></returns>
+        IEnumerator Cooldown(int coolSeconds)
+        {
+            cooldown = true;
```
Comment (frankvHoof93) [inline RPG Wizardry/Assets/Scripts/Avatar/Combat/CastingManager.cs]: Putting cooldown = true outside of the routine makes more sense imo.. (Single Responsibility Principle)

File: RPG Wizardry/Assets/Scripts/PlayerInput/InputState.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+namespace nl.SWEG.RPGWizardry.PlayerInput
+{
+
+    public class InputState : MonoBehaviour
+    {
+        #region Variables
+        #region Public
+
+        //outgoing movement values
+        public Vector3 MovementData;
```
Comment (frankvHoof93) [inline RPG Wizardry/Assets/Scripts/PlayerInput/InputState.cs:16]: use (summary)

**Review by frankvHoof93 — APPROVED** [review]

File: RPG Wizardry/Assets/Scripts/Avatar/Combat/CastingManager.cs (hunk 2 above)
Comment (Thorvers) [reply RPG Wizardry/Assets/Scripts/Avatar/Combat/CastingManager.cs]: tru

File: RPG Wizardry/Assets/Scripts/Avatar/Combat/CastingManager.cs (hunk 1 above)
Comment (Thorvers) [reply RPG Wizardry/Assets/Scripts/Avatar/Combat/CastingManager.cs:16]: to set the projectile that should be fired, yes, i think so

**Pull request merged**